debugger/debugger.go

```diff
@@ -0,0 +1,75 @@
+package debugger
+
+import (
+	"fmt"
+	"net/http"
+	"os/exec"
+
+	"github.com/snes-emu/gose/core"
+)
+
+type Debugger struct {
+	emu  *core.Emulator
+	addr string
+	s    *http.Server
+}
+
+// New debugger instance
+func New(emu *core.Emulator, addr string) *Debugger {
+	db := &Debugger{
+		emu:  emu,
+		addr: addr,
+	}
+	db.createServer(addr)
+
+	return db
+}
+
+//Start the debugger and open the web page
+func (db *Debugger) Start() {
+	go func() {
+		err := db.s.ListenAndServe()
+		fmt.Println(err)
+	}()
+
+	fmt.Printf("open web browser at %s\n", db.addr)
+	cmd := exec.Command("xdg-open", fmt.Sprintf("http://%s", db.addr))
```

Comment:
I think we should first check if `xdg-open` is available (if not we can just log the URL that should be opened or try to fallback on `open`).